ogc/display-ogc-api-collection/src/main/java/com/esri/samples/display_ogc_api_collection/DisplayOgcApiCollectionSample.java

```diff
@@ -0,0 +1,172 @@
+      ogcFeatureCollectionTable = new OgcFeatureCollectionTable(serviceUrl, collectionId);
+
+      // set the feature request mode to manual (only manual is currently supported).
+      // in this mode, the table must be manually populated - panning and zooming won't request features automatically
+      ogcFeatureCollectionTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+      // load the table
+      ogcFeatureCollectionTable.loadAsync();
+
+      // ensure the feature collection table has loaded successfully before creating a feature layer from it to display on the map
+      ogcFeatureCollectionTable.addDoneLoadingListener(() -> {
+        if (ogcFeatureCollectionTable.getLoadStatus() == LoadStatus.LOADED) {
+
+          // create a feature layer and set a renderer to it to visualize the OGC API features
+          var featureLayer = new FeatureLayer(ogcFeatureCollectionTable);
+          var simpleRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, ColorUtil.colorToArgb(Color.BLUE), 3));
+          featureLayer.setRenderer(simpleRenderer);
+
+          // add the layer to the map
+          map.getOperationalLayers().add(featureLayer);
+
+          // zoom to a small area within the dataset by default
+          Envelope datasetExtent = ogcFeatureCollectionTable.getExtent();
+          if (datasetExtent != null && !datasetExtent.isEmpty()) {
+            mapView.setViewpointGeometryAsync(
+              new Envelope(datasetExtent.getCenter(), datasetExtent.getWidth() / 3, datasetExtent.getHeight() / 3));
+          }
+
+        } else {
+          // show an alert dialog if there is a loading failure
+          new Alert(Alert.AlertType.ERROR, "Failed to load OGC Feature Collection Table");
+        }
+      });
+
+      // once the map view navigation has completed, query the OGC API feature table for
+      // additional features within the new visible extent.
+      mapView.addNavigationChangedListener(e -> {
+        if (!e.isNavigating()) {
+
+          // get the current extent
+          Envelope currentExtent = mapView.getVisibleArea().getExtent();
+
+          // create a query based on the current visible extent
+          QueryParameters visibleExtentQuery = new QueryParameters();
+          visibleExtentQuery.setGeometry(currentExtent);
+          visibleExtentQuery.setSpatialRelationship(QueryParameters.SpatialRelationship.INTERSECTS);
+          // set a limit of 5000 on the number of returned features per request, the default on some services
+          // could be as low as 10
+          visibleExtentQuery.setMaxFeatures(5000);
+
+          try {
+            // populate the table with the query, leaving existing table entries intact
+            // setting the outfields parameter to null requests all fields
+            ogcFeatureCollectionTable.populateFromServiceAsync(visibleExtentQuery, false, null);
```

Comment:
You should probably handle the `ListenableFuture` that is returned here - I'm not sure what useful thing you can do with the result though. You could add a done listener and in that call `get()` to see if there are any exceptions... seems a bit complicated though.